File: ProblemSolve/1290.cs
/*****************************************************
 * Convert Binary Number in a Linked List to Integer *
 *****************************************************/

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {
    public int GetDecimalValue(ListNode head) {
        int ans = 0;
        
        //제일 위에 비트부터 계산. 다음 비트가 나오면 x2해서 shift
        //현재 값 추가
        while(head != null){
            ans *= 2;
            ans += head.val;
            head = head.next;
        }
        
        return ans;
    }
}

File: ProblemSolve/1290.cpp
/*****************************************************
 * Convert Binary Number in a Linked List to Integer *
 *****************************************************/

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    int getDecimalValue(ListNode* head) {
        int ans = 0;
        
        //한칸씩 left shift, 새로 들어온 비트 값 추가
        while(head != NULL){
            ans *= 2;
            ans += head -> val;
            head = head -> next;
        }
        
        return ans;
    }
};